### PR TITLE
Add sphinx theme builder & bump to 2022.2.23

### PR DESCRIPTION
**Pull request by gforsyth**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by asmeurer** [comment]
Do we need node as a build requirement?

**Comment by asmeurer** [comment]
No idea. @pradyunsg do you know how to get the build working here? Also we can add you as a feedstock maintainer if you want.

**Comment by pradyunsg** [comment]
You'll need to set STB_USE_SYSTEM_NODE before invoking the build system.

See https://github.com/pradyunsg/sphinx-theme-builder/issues/7#issuecomment-1013820959

**Comment by asmeurer** [comment]
Hmm, the exact version (16.13.1) isn't on conda-forge.

**Comment by asmeurer** [comment]
It is in defaults, though. I tried adding `defaults` as per https://conda-forge.org/docs/maintainer/conda_forge_yml.html?highlight=defaults#channels, but it didn't work. @conda-forge/help-python any idea why this isn't working?

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by carterbox** [comment]
We have `nodejs=16.13.2`, does that version not work?

File: conda-forge.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 conda_forge_output_validation: true
 bot:
   automerge: true
+channels:
+  sources:
+    - conda-forge
+    - defaults
```
Comment (carterbox) [inline conda-forge.yml:7]:
```suggestion
channels:
  sources:
    - conda-forge
    - defaults
build_with_mambabuild: False
```
Also, try using conda-build to see if it respects the channels parameter.

**Comment by asmeurer** [comment]
@carterbox no, see the link in @pradyunsg's comment above. The version has to match exactly or else sphinx-theme-builder won't pick it up. 

We could also try building that version, but it didn't seem straightforward to me. I did notice that 16.13.2 was on anaconda.org, but I didn't find it in the nodejs-feedstock git history. Presumably a 16.13.1 build should reuse the 16.13.2 meta.yaml, since there are many patches in that recipe that would likely not work unless they match the right version.

File: conda-forge.yml (hunk 1 above)
Comment (asmeurer) [reply conda-forge.yml:7]: Sadly it looks like this didn't work.

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

File: conda-forge.yml (hunk 1 above)
Comment (carterbox) [reply conda-forge.yml:7]: Any changes to the conda-forge.yml require a rerender.

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by carterbox** [comment]
@gforsyth, please resolve the merge conflict

@asmeurer, if still doesn't work after conflict resolution and rerender, you can open a PR on https://github.com/conda-forge/nodejs-feedstock/pulls for the missing version.

We should open a bug report to report mambabuild ignoring channels.

**Comment by asmeurer** [comment]
> @gforsyth, please resolve the merge conflict

@carterbox the merge is caused by your suggested change. I'm a little unclear which way it should be resolved. 

> @asmeurer, if still doesn't work after conflict resolution and rerender, you can open a PR on [conda-forge/nodejs-feedstock/pulls](https://github.com/conda-forge/nodejs-feedstock/pulls) for the missing version.

I'd prefer to avoid this for the reasons I mentioned above.

**Comment by carterbox** [comment]
> I'm a little unclear which way it should be resolved.

GitHub doesn't show me conflict because I am not a maintainer of this feedstock.

Unless the conflict is in `conda-forge.yml` or `recipe/`, (on your local machine) please manually merge the upstream branch and then rerender using [conda-smithy](https://github.com/conda-forge/conda-smithy). The merge conflict is probably in one of the files that are manage by conda-smithy.

**Comment by asmeurer** [comment]
@conda-forge-admin rerender

**Comment by asmeurer** [comment]
Fixed the merge conflict. It looks like it is using conda-build, but it still isn't able to find the given version of node.

**Comment by carterbox** [comment]
The issue we're having is documented in https://github.com/conda-forge/conda-smithy/issues/897. The conda-forge.yml documentation is out of date. Please put the channel information in a new conda_build_config.yaml in your recipe directory following step 4 here: https://github.com/conda-forge/conda-smithy#making-a-new-feedstock

Note the syntax of commas with no spaces. This step 4 is also out of date because it looks like defaults is not included by default anymore which is why you will need to explicitly add it.

Something like:

```yaml
channel_sources:
  - conda-forge,defaults
```
Please remember to rerender afterwards, and you can remove all the new additions to conda-forge.yml.

**Comment by asmeurer** [comment]
@conda-forge-admin rerender

**Review by carterbox — CHANGES_REQUESTED** [review]
"recipe/conda_build_config.yaml" check the filename, please.

**Comment by pradyunsg** [comment]
If it makes things easier, I can cut a release with a newer NodeJS requirement in Furo.

**Comment by gforsyth** [comment]
It's green, I'm merging. Thanks to all for getting this through. 

When the upstream version of node is updated we can remove defaults (but no rush @pradyunsg -- greatly appreciate the offer, though)

**Pull request merged**